core/services/relay/evm/mercury/v1/data_source.go

```diff
@@ -292,39 +291,28 @@ func (ds *datasource) executeRun(ctx context.Context) (*pipeline.Run, pipeline.T
 }
 
 func (ds *datasource) setLatestBlocks(ctx context.Context, obs *relaymercuryv1.Observation) {
-	latestBlocks := ds.getLatestBlocks(ctx, nBlocksObservation)
+	latestBlocks, _ := ds.chainReader.LatestHeads(ctx, nBlocksObservation)
```

Comment:
Should this error be checked?

Reply:
The reader on an evm chain always return a nil error, related with the previous review about letting the caller decide if an empty response is an error or not.

https://github.com/smartcontractkit/chainlink/pull/11201/files#diff-237ad2d55c5b2562977f53db7f2e1479118fde028622f031f86f9bf93c69c193R127

Reply:
also the error here is not actually used. we can check for a non nil error on `len(latestBlocks) == 0` and use it as the observation error for consistency.

Reply:
Missed your last comment. Agreed, fixed to use the error on the observation.
CI is failing on a broken import related with the sqlx change.